### PR TITLE
add vllm in LLM module

### PR DESCRIPTION
**Pull request by dahaipeng**
## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Run `pre-commit install` and `pre-commit run --all-files` before git commit, and passed lint check.
* [ ] Some cases need DASHSCOPE_TOKEN_API to pass the Unit Tests, I have at least **pass the Unit tests on local**
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -25,4 +25,5 @@ sentencepiece
 tiktoken
 transformers>=4.33
 unstructured
+vllm
```
Comment (yingdachen) [inline requirements.txt:28]: 我们这些外部依赖，包括langchain，ollama，vllm等等，是不是应该梳理一下，允许optional独立安装

Comment (dahaipeng) [reply requirements.txt:28]: 是的，这里langchain，ollama，vllm是用户使用某些特定功能时才需要安装
UT也这里可能也需要改一下，否则外部依赖不加到requriements.txt中会报错

**Comment by dahaipeng** [comment]
The vllm package is required to pass the unit tests, but there is a question whether it should be put in the requirements.txt of the whole project as it's only needed for vllm users.

**Comment by yingdachen** [comment]
> The vllm package is required to pass the unit tests, but there is a question whether it should be put in the requirements.txt of the whole project as it's only needed for vllm users.

this applies to all requirements that is not core to the project, such as those listed in previous comments. for those requirements, we should be able to setup an environment that include all the requirments, but the depenedency should be setup to be runtime/on-demand dependency.

**Comment by yingdachen** [comment]
this

> > The vllm package is required to pass the unit tests, but there is a question whether it should be put in the requirements.txt of the whole project as it's only needed for vllm users.
> 
> this applies to all requirements that is not core to the project, such as those listed in previous comments. for those requirements, we should be able to setup an environment that include all the requirments, but the depenedency should be setup to be runtime/on-demand dependency.

we do not necessarily need to resolve this in this pr, but at the very least we should open an issue, and plan to sort things out. cc @wenmengzhou @zzhangpurdue

**Comment by dahaipeng** [comment]
> this
> 
> > > The vllm package is required to pass the unit tests, but there is a question whether it should be put in the requirements.txt of the whole project as it's only needed for vllm users.
> > 
> > 
> > this applies to all requirements that is not core to the project, such as those listed in previous comments. for those requirements, we should be able to setup an environment that include all the requirments, but the depenedency should be setup to be runtime/on-demand dependency.
> 
> we do not necessarily need to resolve this in this pr, but at the very least we should open an issue, and plan to sort things out. cc @wenmengzhou @zzhangpurdue

already opened an issue for this problem [[framework] Refactor Non-Core Requirements to Runtime/On-Demand Dependencies](https://github.com/modelscope/modelscope-agent/issues/477)

**Review by zzhangpurdue — APPROVED** [review]

**Pull request merged**